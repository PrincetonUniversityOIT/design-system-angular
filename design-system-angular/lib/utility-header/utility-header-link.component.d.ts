import { RouterLink } from '@angular/router';
/**
 * @example
 `` `
    <jazz-utility-header-link label="Google" url="https://www.google.com/"></jazz-utility-header-link>
 `` `
 */
import * as ɵngcc0 from '@angular/core';
export declare class UtilityHeaderLinkComponent {
    url?: string;
    label: string;
    external: boolean;
    class?: string;
    routerLink: RouterLink | string;
    static ɵfac: ɵngcc0.ɵɵFactoryDeclaration<UtilityHeaderLinkComponent, never>;
    static ɵcmp: ɵngcc0.ɵɵComponentDeclaration<UtilityHeaderLinkComponent, "jazz-utility-header-link", never, { "external": "external"; "url": "url"; "label": "label"; "class": "class"; "routerLink": "routerLink"; }, {}, never, ["*"]>;
}

//# sourceMappingURL=utility-header-link.component.d.ts.map