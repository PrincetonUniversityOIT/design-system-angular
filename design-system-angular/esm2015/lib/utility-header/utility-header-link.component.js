import { Component, Input } from '@angular/core';
/**
 * @example
 `` `
    <jazz-utility-header-link label="Google" url="https://www.google.com/"></jazz-utility-header-link>
 `` `
 */
export class UtilityHeaderLinkComponent {
    constructor() {
        this.external = false;
    }
}
UtilityHeaderLinkComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'jazz-utility-header-link',
                template: `
      <ng-content></ng-content>
  `
            },] }
];
UtilityHeaderLinkComponent.propDecorators = {
    url: [{ type: Input }],
    label: [{ type: Input }],
    external: [{ type: Input }],
    class: [{ type: Input }],
    routerLink: [{ type: Input }]
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidXRpbGl0eS1oZWFkZXItbGluay5jb21wb25lbnQuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi8uLi9wcm9qZWN0cy9kZXNpZ24tc3lzdGVtLWFuZ3VsYXIvc3JjL2xpYi91dGlsaXR5LWhlYWRlci91dGlsaXR5LWhlYWRlci1saW5rLmNvbXBvbmVudC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLEVBQUMsU0FBUyxFQUFFLEtBQUssRUFBQyxNQUFNLGVBQWUsQ0FBQztBQUcvQzs7Ozs7R0FLRztBQVFILE1BQU0sT0FBTywwQkFBMEI7SUFQdkM7UUFVVyxhQUFRLEdBQUcsS0FBSyxDQUFDO0lBRzVCLENBQUM7OztZQWJBLFNBQVMsU0FBQztnQkFDVCw4Q0FBOEM7Z0JBQzlDLFFBQVEsRUFBRSwwQkFBMEI7Z0JBQ3BDLFFBQVEsRUFBRTs7R0FFVDthQUNGOzs7a0JBRUUsS0FBSztvQkFDTCxLQUFLO3VCQUNMLEtBQUs7b0JBQ0wsS0FBSzt5QkFDTCxLQUFLIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHtDb21wb25lbnQsIElucHV0fSBmcm9tICdAYW5ndWxhci9jb3JlJztcbmltcG9ydCB7Um91dGVyTGlua30gZnJvbSAnQGFuZ3VsYXIvcm91dGVyJztcblxuLyoqXG4gKiBAZXhhbXBsZVxuIGBgIGBcbiAgICA8amF6ei11dGlsaXR5LWhlYWRlci1saW5rIGxhYmVsPVwiR29vZ2xlXCIgdXJsPVwiaHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS9cIj48L2phenotdXRpbGl0eS1oZWFkZXItbGluaz5cbiBgYCBgXG4gKi9cbkBDb21wb25lbnQoe1xuICAvLyB0c2xpbnQ6ZGlzYWJsZS1uZXh0LWxpbmU6Y29tcG9uZW50LXNlbGVjdG9yXG4gIHNlbGVjdG9yOiAnamF6ei11dGlsaXR5LWhlYWRlci1saW5rJyxcbiAgdGVtcGxhdGU6IGBcbiAgICAgIDxuZy1jb250ZW50PjwvbmctY29udGVudD5cbiAgYCxcbn0pXG5leHBvcnQgY2xhc3MgVXRpbGl0eUhlYWRlckxpbmtDb21wb25lbnQge1xuICBASW5wdXQoKSB1cmw/OiBzdHJpbmc7XG4gIEBJbnB1dCgpIGxhYmVsOiBzdHJpbmc7XG4gIEBJbnB1dCgpIGV4dGVybmFsID0gZmFsc2U7XG4gIEBJbnB1dCgpIGNsYXNzPzogc3RyaW5nO1xuICBASW5wdXQoKSByb3V0ZXJMaW5rOiBSb3V0ZXJMaW5rIHwgc3RyaW5nO1xufVxuIl19