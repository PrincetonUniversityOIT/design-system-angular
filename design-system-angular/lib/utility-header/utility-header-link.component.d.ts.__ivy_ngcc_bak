import { RouterLink } from '@angular/router';
/**
 * @example
 `` `
    <jazz-utility-header-link label="Google" url="https://www.google.com/"></jazz-utility-header-link>
 `` `
 */
export declare class UtilityHeaderLinkComponent {
    url?: string;
    label: string;
    external: boolean;
    class?: string;
    routerLink: RouterLink | string;
}
