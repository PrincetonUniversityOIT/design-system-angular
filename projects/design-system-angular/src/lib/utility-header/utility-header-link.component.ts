import {Component, Input} from '@angular/core';
import {RouterLink} from '@angular/router';

@Component({
  // tslint:disable-next-line:component-selector
  selector: 'jazz-utility-header-link',
  template: `
      <ng-content></ng-content>
  `,
})
export class UtilityHeaderLinkComponent {
  @Input() url?: string;
  @Input() label: string;
  @Input() external = false;
  @Input() routerLink: RouterLink;
}
