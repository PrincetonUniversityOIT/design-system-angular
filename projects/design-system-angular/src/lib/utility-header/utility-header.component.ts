import {
  Component,
  Input,
  ContentChildren,
  QueryList,
  TemplateRef
} from '@angular/core';
import {UtilityHeaderLinkComponent} from './utility-header-link.component';

@Component({
  // tslint:disable-next-line:component-selector
  selector: 'jazz-utility-header',
  template: `
    <header
        role="banner"
        class="jazz-utility-header"
        [class.jazz-stuck-mobile]="stuckMobile"
        [class.jazz-stuck]="stuckDesktop">
        <h1 class="jazz-visually-hidden">{{ primaryHeading }}</h1>
        <div class="jazz-container">
            <div class="jazz-utility-header__branding">
                <a class="jazz-utility-header__pu_logo" href="https://www.princeton.edu/" [title]="puBrandingTitle">
                <img [src]="puBrandingLogo" alt="Princeton University Logo" />
                </a>
                <div class="jazz-div"></div>
                <div class="jazz-utility-header__site-branding">
                    <a *ngIf="siteBrandingName" class="jazz-utility-header__site-name" [href]="siteBrandingLink" [title]="siteBrandingTitle" rel="home">{{ siteBrandingName }}</a>
                    <div *ngIf="siteBrandingSlogan" class="jazz-utility-header__site-slogan">{{ siteBrandingSlogan }}</div>

                    <ng-template [ngIf]="!officeOfTemplate" [ngIfElse]="officeOfTemplate">
                      <a *ngIf="officeOf" class="jazz-utility-header__site-name-office-of" [href]="officeOfLink" title="Office of {{officeOf}}" rel="home">Office of
                          <br/>
                          <div class="jazz-utility-header__site-name-office-of-department">{{officeOf}}</div>
                      </a>
                    </ng-template>
                </div>
            </div>
            <div class="jazz-utility-header__options">
                <section class="jazz-utility-header__nav" [class.jazz-expanded]="menuExpanded">
                    <h2 class="jazz-visually-hidden">{{ utilityLinksHeading }}</h2>
                    <button class="jazz-utility-header__nav-toggle" [attr.aria-expanded]="menuExpanded" (click)="toggleMenu()"><i class="jazz-icon jazz-icon-menu" [class.jazz-icon-menu]="!menuExpanded" [class.jazz-icon-close]="menuExpanded" aria-hidden="true"></i><span class="jazz-visually-hidden">{{ menuButtonLabel }}</span></button>
                    <nav class="jazz-nav">
                        <ul>
                            <li *ngFor="let link of links"><a [href]="link.url" class="{{link.class}}" [attr.target]="link.external ? '_blank' : null" (click)="toggleMenu()">{{ link.label }}</a></li>
                        </ul>
                    </nav>
                </section>
                <div class="jazz-div"></div>
                <section class="jazz-utility-header__user-options">
                    <h2 class="jazz-visually-hidden">User Options</h2>
                    <ng-template [ngIf]="!userOptionsTemplate" [ngIfElse]="userOptionsTemplate">
                      <ul>
                        <li *ngIf="!username"><a [href]="loginUrl">Log In</a></li>
                        <li *ngIf="username"><a [href]="logoutUrl">Log Out</a></li>
                      </ul>
                    </ng-template>
                </section>
            </div>
        </div>
    </header>
  `
})
export class UtilityHeaderComponent /* implements AfterViewChecked, AfterContentChecked */ {

  @ContentChildren(UtilityHeaderLinkComponent) links: QueryList<UtilityHeaderLinkComponent>;

  @Input('officeOfTemplate')
  officeOfTemplate: TemplateRef<any>;

  @Input('userOptionsTemplate')
  userOptionsTemplate: TemplateRef<any>;

  @Input()
  stuckMobile = true;

  @Input()
  stuckDesktop = true;

  @Input()
  primaryHeading: string;

  @Input()
  puBrandingTitle: string;

  @Input()
  puBrandingLogo = './assets/logos/pu-logo-stacked-white.svg';

  @Input()
  siteBrandingName?: string;

  @Input()
  siteBrandingSlogan?: string;

  @Input()
  siteBrandingLink: string;

  @Input()
  siteBrandingTitle: string;

  @Input()
  officeOfLink?: string;

  @Input()
  officeOf?: string;

  @Input()
  utilityLinksHeading = 'Related Links';

  @Input()
  loginUrl: string;

  @Input()
  logoutUrl: string;

  @Input()
  username: string;

  menuExpanded = false;
  menuButtonLabel = 'Open Navigation Menu';

  toggleMenu(): void {
    this.menuExpanded = !this.menuExpanded;
    this.menuButtonLabel = this.menuExpanded ? 'Close Navigation Menu' : 'Open Navigation Menu';
  }
}
