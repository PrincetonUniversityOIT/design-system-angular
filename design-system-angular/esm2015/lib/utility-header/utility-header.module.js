import { NgModule } from '@angular/core';
import { UtilityHeaderComponent } from './utility-header.component';
import { UtilityHeaderLinkComponent } from './utility-header-link.component';
import { CommonModule } from '@angular/common';
export class UtilityHeaderModule {
}
UtilityHeaderModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    UtilityHeaderComponent,
                    UtilityHeaderLinkComponent
                ],
                imports: [
                    CommonModule
                ],
                exports: [
                    UtilityHeaderComponent,
                    UtilityHeaderLinkComponent
                ]
            },] }
];
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidXRpbGl0eS1oZWFkZXIubW9kdWxlLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiLi4vLi4vLi4vLi4vLi4vcHJvamVjdHMvZGVzaWduLXN5c3RlbS1hbmd1bGFyL3NyYy9saWIvdXRpbGl0eS1oZWFkZXIvdXRpbGl0eS1oZWFkZXIubW9kdWxlLnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBLE9BQU8sRUFBQyxRQUFRLEVBQUMsTUFBTSxlQUFlLENBQUM7QUFDdkMsT0FBTyxFQUFDLHNCQUFzQixFQUFDLE1BQU0sNEJBQTRCLENBQUM7QUFDbEUsT0FBTyxFQUFDLDBCQUEwQixFQUFDLE1BQU0saUNBQWlDLENBQUM7QUFDM0UsT0FBTyxFQUFDLFlBQVksRUFBQyxNQUFNLGlCQUFpQixDQUFDO0FBZTdDLE1BQU0sT0FBTyxtQkFBbUI7OztZQWIvQixRQUFRLFNBQUM7Z0JBQ1IsWUFBWSxFQUFFO29CQUNaLHNCQUFzQjtvQkFDdEIsMEJBQTBCO2lCQUMzQjtnQkFDRCxPQUFPLEVBQUU7b0JBQ1AsWUFBWTtpQkFDYjtnQkFDRCxPQUFPLEVBQUU7b0JBQ1Asc0JBQXNCO29CQUN0QiwwQkFBMEI7aUJBQzNCO2FBQ0YiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQge05nTW9kdWxlfSBmcm9tICdAYW5ndWxhci9jb3JlJztcbmltcG9ydCB7VXRpbGl0eUhlYWRlckNvbXBvbmVudH0gZnJvbSAnLi91dGlsaXR5LWhlYWRlci5jb21wb25lbnQnO1xuaW1wb3J0IHtVdGlsaXR5SGVhZGVyTGlua0NvbXBvbmVudH0gZnJvbSAnLi91dGlsaXR5LWhlYWRlci1saW5rLmNvbXBvbmVudCc7XG5pbXBvcnQge0NvbW1vbk1vZHVsZX0gZnJvbSAnQGFuZ3VsYXIvY29tbW9uJztcblxuQE5nTW9kdWxlKHtcbiAgZGVjbGFyYXRpb25zOiBbXG4gICAgVXRpbGl0eUhlYWRlckNvbXBvbmVudCxcbiAgICBVdGlsaXR5SGVhZGVyTGlua0NvbXBvbmVudFxuICBdLFxuICBpbXBvcnRzOiBbXG4gICAgQ29tbW9uTW9kdWxlXG4gIF0sXG4gIGV4cG9ydHM6IFtcbiAgICBVdGlsaXR5SGVhZGVyQ29tcG9uZW50LFxuICAgIFV0aWxpdHlIZWFkZXJMaW5rQ29tcG9uZW50XG4gIF1cbn0pXG5leHBvcnQgY2xhc3MgVXRpbGl0eUhlYWRlck1vZHVsZSB7XG5cbn1cbiJdfQ==