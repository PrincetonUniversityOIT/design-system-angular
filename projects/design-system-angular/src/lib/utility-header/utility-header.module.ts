import {NgModule} from '@angular/core';
import {UtilityHeaderComponent} from './utility-header.component';
import {UtilityHeaderLinkComponent} from './utility-header-link.component';
import {CommonModule} from '@angular/common';

@NgModule({
  declarations: [
    UtilityHeaderComponent,
    UtilityHeaderLinkComponent
  ],
  imports: [
    CommonModule
  ],
  exports: [
    UtilityHeaderComponent,
    UtilityHeaderLinkComponent
  ]
})
export class UtilityHeaderModule {

}
