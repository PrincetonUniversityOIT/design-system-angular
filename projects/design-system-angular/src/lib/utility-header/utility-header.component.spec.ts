import { ComponentFixture, TestBed } from '@angular/core/testing';

import { UtilityHeaderComponent } from './utility-header.component';
import { UtilityHeaderLinkComponent} from './utility-header-link.component';
import {Component} from '@angular/core';

@Component({
  selector: 'lib-links-mock',
  template: `
    <jazz-utility-header>
      <jazz-utility-header-link label="Link 1" url="http://link1/"></jazz-utility-header-link>
      <jazz-utility-header-link label="Link 2" url="http://link2/" *ngIf="false"></jazz-utility-header-link>
      <jazz-utility-header-link label="Link 3" url="http://link3/" [external]="true"></jazz-utility-header-link>
    </jazz-utility-header>
  `
})
export class TestLinkMockComponent { }

describe('UtilityHeaderComponent', () => {
  let component: UtilityHeaderComponent;
  let fixture: ComponentFixture<UtilityHeaderComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        UtilityHeaderComponent,
        UtilityHeaderLinkComponent,
        TestLinkMockComponent
      ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(UtilityHeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should set jazz-stuck and jazz-stuck-mobile by default', () => {
    const header = fixture.nativeElement.querySelector('header.jazz-utility-header');
    expect(header).toHaveClass('jazz-stuck');
    expect(header).toHaveClass('jazz-stuck-mobile');
  });

  it('should not set jazz-stuck and jazz-stuck-mobile when disabled', () => {
    component.stuckDesktop = false;
    component.stuckMobile = false;
    fixture.detectChanges();
    const header = fixture.nativeElement.querySelector('header.jazz-utility-header');
    expect(header).not.toHaveClass('jazz-stuck');
    expect(header).not.toHaveClass('jazz-stuck-mobile');
  });

  it('should open and close menu when toggle button is clicked', () => {
    const button = fixture.nativeElement.querySelector('button.jazz-utility-header__nav-toggle');
    expect(button).toHaveAttribute('aria-expanded', 'false');
    const buttonIcon = button.querySelector('i.jazz-icon');
    expect(buttonIcon).toHaveClass('jazz-icon-menu');
    const buttonSpan = button.querySelector('span');
    expect(buttonSpan).toHaveTextContent('Open Navigation Menu');
    const section = fixture.nativeElement.querySelector('section.jazz-utility-header__nav');
    expect(section).not.toHaveClass('jazz-expanded');

    // open the menu with a click

    button.click();
    fixture.detectChanges();
    expect(section).toHaveClass('jazz-expanded');
    expect(button).toHaveAttribute('aria-expanded', 'true');
    expect(buttonIcon).toHaveClass('jazz-icon-close');
    expect(buttonSpan).toHaveTextContent('Close Navigation Menu');

    // close the menu with a click

    button.click();
    fixture.detectChanges();
    expect(section).not.toHaveClass('jazz-expanded');
    expect(button).toHaveAttribute('aria-expanded', 'false');
    expect(buttonIcon).toHaveClass('jazz-icon-menu');
    expect(buttonSpan).toHaveTextContent('Open Navigation Menu');
  });
});

describe('UtilityHeaderComponentTemplate', () => {
  let component: TestLinkMockComponent;
  let fixture: ComponentFixture<TestLinkMockComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        UtilityHeaderComponent,
        UtilityHeaderLinkComponent,
        TestLinkMockComponent
      ]
    })
      .compileComponents();
  });

  it('should translate link tags into links', () => {
    fixture = TestBed.createComponent(TestLinkMockComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();

    const links = fixture.nativeElement.querySelectorAll('section.jazz-utility-header__nav > nav.jazz-nav > ul > li > a');
    // the middle link will be hidden by an *ngIf
    expect(links.length).toEqual(2);
    expect(links[0]).toHaveAttribute('href', 'http://link1/');
    expect(links[0]).toHaveTextContent('Link 1');
    expect(links[0]).not.toHaveAttribute('target');
    expect(links[1]).toHaveAttribute('href', 'http://link3/');
    expect(links[1]).toHaveTextContent('Link 3');
    expect(links[1]).toHaveAttribute('target', '_blank');
  });
});
