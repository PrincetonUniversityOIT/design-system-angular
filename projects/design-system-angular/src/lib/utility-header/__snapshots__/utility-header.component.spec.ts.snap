// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UtilityHeaderComponentOfficeOfTemplate should put office of template into the office of section of the header 1`] = `
<a
  class="jazz-utility-header__site-name-office-of"
  href="https://www.google.com"
  rel="home"
  title="Dept of Energy"
>
  Department of 
  <br />
  <div
    class="jazz-utility-header__site-name-office-of-department"
  >
    Energy
  </div>
</a>
`;
